### DISCUSSION TITLE
Could someone explain `requirements` a bit more?

### DISCUSSION THREAD

**Discussion by jonathanj** [microsoft/rnx-kit#2340] [Q&A] [answered]
The [requirements documentation](https://microsoft.github.io/rnx-kit/docs/tools/align-deps#requirements) mentions that "microsoft/react-native/0.70 will be used for development", in the context of the example earlier in the docs. But it's not clear what `microsoft/react-native/0.70` is in this case.

Does it refer to [this profile declared in the presets](https://github.com/microsoft/rnx-kit/blob/f74073891bc9417ab4b39dd47655ee2a491b3c40/packages/align-deps/src/presets/microsoft/react-native.ts#L27)?

What am trying to do is use align-deps for a non-RN (utility) package in a monorepo containing RN apps, but I can't omit `requirements` so I'm trying to understand how to correctly define it when the package does not (and should not) depend on RN itself.

Would greatly appreciate any advice!

**Comment by tido64** [comment, marked as answer]
Hi @jonathanj.

> Does it refer to [this profile declared in the presets](https://github.com/microsoft/rnx-kit/blob/f74073891bc9417ab4b39dd47655ee2a491b3c40/packages/align-deps/src/presets/microsoft/react-native.ts#L27)?

Yes, that's correct. Specifically, this profile: [`packages/align-deps/src/presets/microsoft/react-native/profile-0.70.ts`](https://github.com/microsoft/rnx-kit/blob/f74073891bc9417ab4b39dd47655ee2a491b3c40/packages/align-deps/src/presets/microsoft/react-native/profile-0.70.ts)

> What am trying to do is use align-deps for a non-RN (utility) package in a monorepo containing RN apps, but I can't omit `requirements` so I'm trying to understand how to correctly define it when the package does not (and should not) depend on RN itself.

In the `requirements` section, you specify that your package needs. Instead of `react-native@0.70`, you could state `react@17` if your package requires React 17.x.

I hope that makes sense.

**Reply by jonathanj** [reply]
Thanks for the reply @tido64, that does help clear it up, thanks.

I have a follow-up question then, out of curiosity: `requirements` as an arbitrary package name that my package needs (like in the most fundamental sense) seems sort of arbitrary to me. From my (a user's) point of view, this could also be achieved as a capability? So I assume that `requirements` is useful to align-deps in some other way, I'm just not sure what that way is, would you mind explaining it or pointing me to the documentation I may have missed on the subject?

**Reply by tido64** [reply]
The design document we wrote contains a lot more details: https://microsoft.github.io/rnx-kit/docs/architecture/dependency-management

It still uses the old terminologies from 1.x, but I still think you should be able to grasp some of the basic concepts. Let me know if things are still unclear after reading it. We can always improve our docs.

**Reply by jonathanj** [reply]
Thanks, that makes it clearer to me. I appreciate the feedback!
